### PR TITLE
SI-807: Exit on build container fail

### PR DESCRIPTION
**Pull request by ravid-stratoscale**

### CODE REVIEW

File: skipper/cli.py [hunk 1]
```diff
@@ -258,7 +258,11 @@ def _prepare_build_container(registry, image, tag, git_revision=False):
 
     docker_file = utils.image_to_dockerfile(image)
     utils.logger.info("Building image using docker file: %(docker_file)s", dict(docker_file=docker_file))
-    runner.run(['docker', 'build', '-t', image, '-f', docker_file, '.'])
+    ret = runner.run(['docker', 'build', '-t', image, '-f', docker_file, '.'])
+
+    if ret != 0:
+        utils.logger.error('Failed to build image: %(image)s', dict(image=image))
+        raise Exception('Failed to build image: %(image)s', dict(image=image))
```
Comment (shay-stratoscale) [inline skipper/cli.py]: `RuntimeError` is more suited for this case..

Comment (shay-stratoscale) [inline skipper/cli.py]: do we really need both log & exception?

**Comment by ravid-stratoscale** [comment]
@shay-stratoscale , look now

File: skipper/cli.py [hunk 2]
```diff
@@ -258,7 +274,10 @@ def _prepare_build_container(registry, image, tag, git_revision=False):
 
     docker_file = utils.image_to_dockerfile(image)
     utils.logger.info("Building image using docker file: %(docker_file)s", dict(docker_file=docker_file))
-    runner.run(['docker', 'build', '-t', image, '-f', docker_file, '.'])
+    ret = runner.run(['docker', 'build', '-t', image, '-f', docker_file, '.'])
+
+    if ret != 0:
+        raise RuntimeError('Failed to build image: %(image)s', dict(image=image))
```
Comment (shay-stratoscale) [inline skipper/cli.py]: how about removing the exception and writing

`exit('Failed to build image: %(image)s', dict(image=image))`

the program will exit immediately with the proper message, so you wouldn't have to manage the exceptions and logs...

Comment (shay-stratoscale) [inline skipper/cli.py]: you an also use an `asset` statement

**Pull request merged**